private_routing.tf

```diff
@@ -135,5 +135,6 @@ resource "aws_route" "nat_gateway" {
   depends_on = [
     var.module_depends_on,
     aws_nat_gateway.nat_gateway,
+    aws_route_table.private
```

Comment:
why? a nat gateway does not need a route table to exist before it is created!?

Reply:
> ping @soerenmartius any progress on this PR?

no time at the moment - will convert back to a draft and address once we plan this accordingly. The dependencies didn't resolve properly btw that's why I added them to `depends_on`